src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -545,7 +545,23 @@ internal static string GetPipeNameForPathOpt(string compilerExeDirectory)
                 return null;
             }
 
-            return $"{userName}.{(isAdmin ? 'T' : 'F')}.{basePipeName}";
+            var pipeName = $"{userName}.{(isAdmin ? 'T' : 'F')}.{basePipeName}";
+
+#if NETCOREAPP2_0
+            if (!PlatformInformation.IsWindows)
+            {
+                // Check if the pipe's path would be longer than a legal path on Mac
+                var getPipePath = typeof(PipeStream).GetMethod("GetPipePath", BindingFlags.Static | BindingFlags.NonPublic);
+                var pipePath = (string)getPipePath.Invoke(null, new[] { ".", pipeName });
+                const int MaxPipePathLength = 100;
```

Comment:
@stephentoub last email recommended we use 92 here as it's the most portable value. Also we should document this magic number.

Reply:
Yup. That comes from the man page which states "When coding portable applications, keep in mind that some implementations have sun_path as short as 92 bytes." Last I checked, macOS was close to that min.

Reply:
Which man page has this information? That would be good to list here in a comment.

Reply:
e.g. http://man7.org/linux/man-pages/man7/unix.7.html